### DISCUSSION TITLE
How to migrate from Docker PULL + TAG + PUSH commands to Crane equivalents?

### DISCUSSION THREAD

**Discussion by simon-liebehenschel** [gocrane/crane#550] [Q&A]
In my CI/CD I pull an image from the private registry, change its tag, and push the image to Google Cloud.

What I do now:
```yml
    - docker pull gitlab/IMAGE
    - docker tag gitlab/IMAGE eu.gcr.io/IMAGE
    - docker push eu.gcr.io/IMAGE
```
The problem is that Crane has different logic for the same commands, so I can not just `crane pull`, etc. For example, Crane accepts two arguments for the `pull`.

`crane pull IMAGE TARBALL [flags]`   I have no idea what I need to write as a "tarball" argument in my case.

What is the correct way to migrate from Docker (PULL + TAG + PUSH workflow) to the Crane to achieve the same result?

**Comment by mfanjie** [comment]
I did not understand your question.
I suppose you should be able to pull image by `docker pull gocrane/craned:v0.7.0` and tag it by
`docker tag gocrane/craned:v0.7.0 eu.gcr.io/craned:v0.7.0` and then push it by `docker push eu.gcr.io/craned:v0.7.0`

**Reply by simon-liebehenschel** [reply]
No, I must not use Docker. I must use `crane` to work with my images. 

The problem is that I do not understand how to use `crane pull` in my CI/CD because it is not clear what arguments it wants. I mean that `docker pull` and `crane pull` accept different arguments, and it is not clear what exactly I need to specify as a `TARBALL`.

E.g. if I have `docker pull my-private-registry/my-image:latest`, then how will look `crane` equivalent to achieve the same?
